### ISSUE TITLE
Support for Safari

### ISSUE THREAD

**Issue by nncl** [MSakamaki/ngx-onesignal#48]
#### I am submitting a

- [x] Bug Report

#### What is the expected behavior?
Be able to get the user id through `await OneSignal.getUserId()`

#### What is the current behavior?
It returns a bug when subscribing as shown:

![Screen Shot 2020-10-31 at 4 38 19 PM](https://user-images.githubusercontent.com/6740387/97788431-a5c38d00-1b97-11eb-9cae-3a9f3bcecd63.png)
![Screen Shot 2020-10-31 at 4 37 57 PM](https://user-images.githubusercontent.com/6740387/97788433-a78d5080-1b97-11eb-9b9d-dccac0c1668e.png)

#### What are the steps to reproduce?
As far as I can see just creating the project as the example in the docs.

#### Which versions are you using for the following packages?

Angular: 10.2.0
Angular CDK: 10.2.6
Angular CLI: 10.2.0
Nx (option):
ngx-onesignal: 10.0.0

#### Is there anything else we should know?

This is my **app.module**:

```typescript
NgxOneSignalModule.forRoot({
      appId: 'my-app-id,
      autoRegister: false,
      allowLocalhostAsSecureOrigin: true,
      notifyButton: {
        enable: true,
      },
    }),
```

And my **app.component**:

```typescript
import { Component, OnInit } from '@angular/core';
import { OneSignalService } from 'ngx-onesignal';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent implements OnInit {

  constructor(public readonly onesignal: OneSignalService) {
    (window as any).ngxOnesignal = this.onesignal;
  }

  ngOnInit(): void {
    setTimeout(() => {
      this.onesignal.subscribe();
      console.log(`Initiated`);
    }, 2000);
  }
}
```

Also, I couldn't see anything about the `safari_web_id` property as described in https://documentation.onesignal.com/docs/safari-web-push-setup, so, does this wrapper supports it?

I really appreciate your help :)

**Comment by MSakamaki** [comment]
@nncl 
Yes, I support safari_safari_web_id in safari.

I have been suffering from the same phenomenon since the day before yesterday, but now it seems to be able to call the API successfully.
Are still experiencing the problem in your environment?
If the problem is still there, I'll find a reproducible environment and look deeper into it.

**Comment by nncl** [comment]
@MSakamaki yes, unfortunately still happening when testing on Safari this following environment https://onesignal-testing.surge.sh/ Have you tried? If so, this error is shown for you?

Thanks for your time.

**Comment by MSakamaki** [comment]
Temporary Report

After investigating, I found that there is a problem with the Driver inside ngsw-worker.js.

It appears that the server is returning a CORS error due to the server hooking up the "https://onesignal.com/" communication when fetching onesignal communications and executing from the application's domain.
We have been able to confirm that it can be worked around by temporarily disabling this feature, but we are still investigating the underlying solution.

https://github.com/angular/angular/blob/master/packages/service-worker/worker/src/driver.ts#L404

**Comment by nncl** [comment]
@MSakamaki thanks, I appreciate your time. I'll try to disable and test as you did, and also if I come up with something else I'll let you know.
